filebeat/crawler/prospector_log.go

```diff
@@ -228,6 +228,8 @@ func (p *ProspectorLog) checkExistingFile(h *harvester.Harvester, newFile *input
 
 			// Start a new harvester on the path
 			h.Start()
+			p.Prospector.registrar.Persist <- h.GetState()
```

Comment:
Well, h.GetState returns the state as known to harvester. Not state known to registrar or publisher. h.GetState might be well ahead of registrar as h.GetState() is harvesters internal state into how many bytes have I read. The registrars state on the other hand should be based on: how many bytes have I reported. If publish fails or takes longer, the harvesters state is well ahead of state in registrar. Not sure how likely this scenario is though